lib/transactions.js

```diff
@@ -216,7 +216,45 @@ exports.createGeneration = (rpcData, blockReward, feeReward, recipients, poolAdd
                 Math.round(blockReward.total * (1 - (feePercent / 100))) + feeReward
             )
         }
-    } else {
+    
+    } else if(rpcData.coinbase_required_outputs && rpcData.coinbase_required_outputs.length){
+     /* This section is for ANON (Anonymous Bitcoin)   
+     // ANON getblocktemplate provides an array of objects (coinbase_required_outputs) 
+     // it contains all the required coinbase outputs (masternode, developement fund, superblocks)
+     // each object has the following keys: amount, script (hex), type.
+     // 'type' provides short info about the purpose of the output
+     // for example 'masternode', 'development' or 'superblock'
+    */
+     
+     // keep track of total the amount of all outputs (except superblock) in coinbase_required_outputs array
+     let required_outputs_total = 0; 
+        // loop through coinbase_required_outputs and add them to our coinbase transaction
+        for(let i = 0; i < rpcData.coinbase_required_outputs.length; i++){
+
+            let script = new Buffer(rpcData.coinbase_required_outputs[i].script, 'hex')
+            let amount = rpcData.coinbase_required_outputs[i].amount;
+            
+            // let's find out if it's a superblock 
+            // we ignore superblock's output value becuase it's not part of the block reward
+            // during superblocks the total output value of coinbase exceeds regular block reward
+            if(rpcData.coinbase_required_outputs[i].type !== "superblock")
+                required_outputs_total += amount;
+            
+            txb.addOutput(
+                script,
+                amount
+            )
+        }
+        // we want to calculate pool fee using miner reward only
+        blockReward.total = blockReward.total - required_outputs_total;
```

Comment:
This can be simplified to: `blockReward.total -= required_outputs_total`

Reply:
Fixed.